browser/ui/views/tabs/brave_tab_container.cc

```diff
@@ -444,22 +463,185 @@ BrowserRootView::DropIndex BraveTabContainer::GetDropIndex(
+    TabContainerImpl::HandleDragExited();
     return;
   }
-  TabContainerImpl::HandleDragExited();
+  SetDropArrow({});
+}
+
+gfx::Rect BraveTabContainer::GetDropBounds(int drop_index,
+                                           bool drop_before,
+                                           bool drop_in_group,
+                                           bool* is_beneath) {
+  DCHECK_NE(drop_index, -1);
+
+  // The center is determined along the x-axis if it's pinned, or along the
+  // y-axis if not.
+  int center = -1;
+
+  if (GetTabCount() == 0) {
+    // If the tabstrip is empty, it doesn't matter where the drop arrow goes.
+    // The tabstrip can only be transiently empty, e.g. during shutdown.
+    return gfx::Rect();
+  }
+
+  Tab* tab = GetTabAtModelIndex(std::min(drop_index, GetTabCount() - 1));
+
+  const bool is_tab_pinned = tab->data().pinned;
+
+  const bool first_in_group =
+      drop_index < GetTabCount() && tab->group().has_value() &&
+      GetModelIndexOf(tab) ==
+          controller_->GetFirstTabInGroup(tab->group().value());
+
+  if (!drop_before || !first_in_group || drop_in_group) {
+    // Dropping between tabs, or between a group header and the group's first
+    // tab.
+    center = is_tab_pinned ? tab->x() : tab->y();
+    const int length = is_tab_pinned ? tab->width() : tab->height();
+    if (drop_index < GetTabCount()) {
+      center += drop_before ? -(tabs::kVerticalTabsSpacing / 2) : (length / 2);
+    } else {
+      center += length + (tabs::kVerticalTabsSpacing / 2);
+    }
+  } else {
+    // Dropping before a group header.
+    TabGroupHeader* const header = group_views_[tab->group().value()]->header();
+    // Since there is no tab group in pinned tabs, there is no need to consider
+    // the x-axis.
+    center = header->y() + tabs::kVerticalTabsSpacing / 2;
+  }
+
+  // Determine the screen bounds.
+  gfx::Point drop_loc(is_tab_pinned ? center - g_drop_indicator_width / 2 : 0,
+                      is_tab_pinned ? tab->y() - g_drop_indicator_height
+                                    : center - g_drop_indicator_height / 2);
+  ConvertPointToScreen(this, &drop_loc);
+  gfx::Rect drop_bounds(drop_loc.x(), drop_loc.y(), g_drop_indicator_width,
+                        g_drop_indicator_height);
+
+  // If the rect doesn't fit on the monitor, push the arrow to the bottom.
+  display::Screen* screen = display::Screen::GetScreen();
+  display::Display display = screen->GetDisplayMatching(drop_bounds);
+  *is_beneath = !display.bounds().Contains(drop_bounds);
+
+  if (*is_beneath) {
+    drop_bounds.Offset(
+        is_tab_pinned ? 0 : drop_bounds.width() + tab->width(),
+        is_tab_pinned ? drop_bounds.height() + tab->height() : 0);
+  }
+
+  return drop_bounds;
+}
+
+gfx::ImageSkia* BraveTabContainer::GetDropArrowImage(
+    BraveTabContainer::DropArrow::Position pos,
+    bool beneath) {
+  return ui::ResourceBundle::GetSharedInstance().GetImageSkiaNamed(
```

Comment:
When you transform the existing image, caching transformed images in local `static base::NoDestructor` would be great. Then we'll also be able to remove g_arrow_width/height.